### ISSUE TITLE
IRAM 30m FITS file with SFL direction coordinate and corresponding PV header keywords causes wcsset_error

### ISSUE THREAD

**Issue by dpetry** [casacore/casacore#1317]
A recent single-dish image in FITS format cannot properly be imported into a CASA image under CASA 6.5 .
The images uses the SFL projection with the (marginally standard) additional header keywords PVi_0 - PVi_2
to specify a non-trivial celestial reference point (PVi_1 and PVi_2) and whether that point is to be used (PVi_0 == 1).
The root cause of the error seems to be an unawareness of this special regulation for SFL 
in the code in coordinates/Coordinates/FITSCoordinateUtil.cc and related files.
Need to hunt down the exact place where the fix needs to be applied.

The IRAM image passes the fitsverify test without problems, and so the casacore code should be able to read it.
I.e. this is really a casacore bug in the Coordinates section.

**Comment by dpetry** [comment]
@tammojan , there have been many new casacore team members in recent years. Can you tell me if any one of them has specialised in the FITS part of the Coordinates code, or am I still the last person to have touched this code?

**Comment by tammojan** [comment]
I'm quite sure that this code has not been touched significantly by others. The latest meaningful commit to FITSCoordinateUtil.cc was 72a630014e5b549d86496d7e7815b3c820f71922 by you in 2015. That change was also about SFL with reference points set.

**Comment by dpetry** [comment]
@tammojan , thanks. I will then work on improving it further. This Sanson Flamsteed projection is something which was not in the FIT standard originally but added later and it is apparently quite popular with some people ...

**Comment by dpetry** [comment]
It turns out the the IRAM30M FITS file which I tried to read here,
is not using correct FITS spatial coordinate projection syntax.

CRPIX1  =       8.285713918178 / Pixel coordinate of reference point
CRPIX2  =       5.571428693941 / Pixel coordinate of reference point
CRPIX3  =                  1.0 / Pixel coordinate of reference point
CDELT1  =   -0.003888888932876 / [deg] Coordinate increment at reference point
CDELT2  =    0.003888888932876 / [deg] Coordinate increment at reference point
CDELT3  =                  0.2 / [m/s] Coordinate increment at reference point
CUNIT1  = 'deg'                / Units of coordinate increment and value
CUNIT2  = 'deg'                / Units of coordinate increment and value
CUNIT3  = 'km s-1  '           / Units of coordinate increment and value
CTYPE1  = 'RA---SFL'           / Right ascension, Sanson-Flamsteed projection
CTYPE2  = 'DEC--SFL'           / Declination, Sanson-Flamsteed projection
CTYPE3  = 'VRAD'               / Radio velocity (linear)
CRVAL1  =       276.5779166667 / [deg] Coordinate value at reference point
CRVAL2  =      -12.68786111111 / [deg] Coordinate value at reference point
CRVAL3  =      35.204488164675 / [m/s] Coordinate value at reference point
PV1_0   =                  1.0 / [m/s] Coordinate value at reference point
PV1_1   =                  0.0 / [deg] Native longitude of the reference point
PV1_2   =      -12.68786111111 / [deg] Native latitude  of the reference point
LONPOLE =                  0.0 / [deg] Native longitude of celestial pole
LATPOLE =                 90.0 / [deg] Native latitude of celestial pole

As one can see, the image uses the SFL projection.
According to table 13 of Calabretta & Greisen 2002, this projection has no
additional PV parameters (as opposed to, e.g., BON which uses PV1_1)!
However, this header contains PV1_0, PV1_1, and PV1_2
and PV1_0 is non-zero, i.e. these parameters are "activated".

This causes the FITS reader in casacore (which uses wcslib 7 at the moment) to fail.
And rightly so, because the three PV values should not be there
or at least PV1_0 should be zero. 

When I set PV1_0 to zero in the above header and try to read the image with CASA,
there is no error.

So, this is a non-issue. The creators of the FITS file made a mistake.
